cli/flox/src/commands/init/python.rs

```diff
@@ -662,3 +671,236 @@ impl Display for ProvidedVersion {
+        [project]
+        requires-python = ">=3.8"
+        "#};
+
+        let pyproject = PyProject::from_pyproject_content(content, flox).unwrap();
+
+        assert_eq!(pyproject.unwrap(), PyProject {
+            provided_python_version: ProvidedVersion::Compatible {
+                requested: Some(">=3.8".to_string()),
+                compatible: "3.11.6".to_string(),
+            },
+            pip_version: "23.2.1".to_string(),
+        });
+    }
+
+    /// ProvidedVersion::Incompatible should be returned for requires-python = "1"
+    #[test]
+    #[serial]
+    fn test_pyproject_unavailable_version() {
+        let flox = &FLOX_INSTANCE.0;
+
+        // TODO: python docs have a space in the version (>= 3.8)
+        // https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires
+        // pkgdb currently throws an exception when passed that specifier
+        let content = indoc! {r#"
+        [project]
+        requires-python = "1"
+        "#};
+
+        let pyproject = PyProject::from_pyproject_content(content, flox).unwrap();
+
+        assert_eq!(pyproject.unwrap(), PyProject {
+            provided_python_version: ProvidedVersion::Incompatible {
+                requested: "1".to_string(),
+                substitute: "3.11.6".to_string(),
+            },
+            pip_version: "23.2.1".to_string(),
+        });
+    }
+
+    /// An invalid pyproject.toml should return an error
+    #[test]
+    fn test_poetry_pyproject_invalid() {
+        let flox = &FLOX_INSTANCE.0;
+
+        let content = indoc! {r#"
+        ,
+        "#};
+
+        let pyproject = PoetryPyProject::from_pyproject_content(content, flox);
+
+        assert!(pyproject.is_err());
+    }
+
+    /// None should be returned for an empty pyproject.toml
+    #[test]
+    fn test_poetry_pyproject_empty() {
+        let flox = &FLOX_INSTANCE.0;
+
+        let pyproject = PoetryPyProject::from_pyproject_content("", flox).unwrap();
+
+        assert_eq!(pyproject, None);
+    }
+
+    /// Err should be returned for a pyproject.toml with `tool.poetry` but not
+    /// `tool.poetry.dependencies.python`
+    #[test]
+    fn test_poetry_pyproject_no_python() {
+        let flox = &FLOX_INSTANCE.0;
+
+        let content = indoc! {r#"
+        [tool.poetry]
+        "#};
+
+        let pyproject = PoetryPyProject::from_pyproject_content(content, flox);
+
+        assert!(pyproject.is_err());
+    }
+
+    /// ProvidedVersion::Compatible should be returned for python = "^3.7"
+    #[test]
+    #[serial]
+    fn test_poetry_pyproject_available_version() {
+        let flox = &FLOX_INSTANCE.0;
+
+        // TODO: python docs have a space in the version (>= 3.8)
+        // https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires
+        // pkgdb currently throws an exception when passed that specifier
+        let content = indoc! {r#"
+        [tool.poetry.dependencies]
+        python = "^3.7"
+        "#};
```

Comment:
poetry seem to adhere to standard semver: https://python-poetry.org/docs/dependency-specification#caret-requirements

though they might be more lenient with it so fixing it for both cases seems reasonable anyway